apps/prairielearn/src/batched-migrations/20250215015626_course_instance_usages__workspaces__backfill.ts

```diff
@@ -0,0 +1,26 @@
+import { z } from 'zod';
+
+import { makeBatchedMigration } from '@prairielearn/migrations';
+import { loadSqlEquiv, queryAsync, queryRow } from '@prairielearn/postgres';
+
+const sql = loadSqlEquiv(import.meta.url);
+
+const CUTOFF_DATE = '2025-02-15T00:00:00Z';
```

Comment:
I hope that this date is now sufficiently far in the past that we get more or less perfect data out of the backfill (not that it _really_ matters).

Reply:
We're still going to miss a small amount of data. The live-updating query updates every time we have a state change, while the backfill only considers workspaces where the last update is before the cutoff date.

Consider a workspace started before the cutoff date, which was used for 12 hours and then stopped. Later it was started up again after the cutoff date and then used for a further 12 hours. The live-updating query will only catch the second 12-hour period, and the backfill job won't process this workspace at all, so the first 12-hours is lost.

This should only affect workspaces started before the cutoff and re-used after, which is hopefully a very small fraction overall.